### PR TITLE
OCPBUGS-38479: ic: validate release arch is compatible with cluster arch

### PR DESCRIPTION
**Pull request by r4f4**
Check that the release payload can be used to create nodes from the specified architecture during install-config validation. If we don't do that, the install will fail with the error:
```
ERROR Attempted to gather ClusterOperator status after installation failure: listing ClusterOperator objects: Get "https://api.build11.ci.devcluster.openshift.com:6443/apis/config.openshift.io/v1/clusteroperators": dial tcp 13.59.207.137:6443: connect: connection refused
```
while the real reason is in the bootstrap bootkube logs:
```
Release image arch amd64 does not match host arch arm64
```

### CODE REVIEW

**Comment by r4f4** [comment]
/cc @jeffdyoung 
It would be nice to have a pair of :eyes:  from the multi-arch team

**Review by barbacbd — COMMENTED** [review]
/lgtm

**Comment by r4f4** [comment]
Testing this with the release payload built for the aws CI job so it contains the installer with this PR changes:
```bash
$ ./openshift-install version
./openshift-install 4.18.0-0.ci.test-2024-08-16-174208-ci-op-95knv4ri-latest
built from commit 4b36a57f97107fc3b8fcec8e642bb7608ea2294e
release image registry.build01.ci.openshift.org/ci-op-95knv4ri/release@sha256:1a8555a6a3c70a5d514fae264dfc50514226b5da8c294e9ca9334b8a50dc9d29
release architecture amd64
$ yq 'del(.pullSecret,.sshKey)' install-config.yaml
```
```yaml
apiVersion: v1
metadata:
  name: rdossant-installer-08141525
publish: External
baseDomain: devcluster.openshift.com
platform:
  aws:
    region: us-east-2
compute:
  - name: worker
    architecture: arm64
controlPlane:
  name: master
  architecture: arm64
```
```bash
$ ./openshift-install create manifests
ERROR failed to fetch Master Machines: failed to load asset "Install Config": failed to create install config: invalid "install-config.yaml" file: [controlPlane.architecture: Invalid value: "arm64": cannot create arm64 controlPlane node from a single architecture amd64 release payload, compute[0].architecture: Invalid value: "arm64": cannot create arm64 compute node from a single architecture amd64 release payload]
$ yq --inplace '.controlPlane.architecture = "amd64"' install-config.yaml
$ ./openshift-install create manifests
ERROR failed to fetch Master Machines: failed to load asset "Install Config": failed to create install config: invalid "install-config.yaml" file: compute[0].architecture: Invalid value: "arm64": cannot create arm64 compute node from a single architecture amd64 release payload
$ yq --inplace '.compute[0].architecture = "amd64"' install-config.yaml
$ ./openshift-install create manifests
WARNING Length of cluster name "rdossant-installer-08141525" is 27 which is greater than the max 21 allowed. The name will be truncated to "rdossant-installer-08"
INFO Consuming Install Config from target directory
INFO Adding clusters...
INFO Manifests created in: cluster-api, manifests and openshift
```

File: pkg/types/validation/installconfig.go [hunk 1]
```diff
@@ -125,11 +125,17 @@ func ValidateInstallConfig(c *types.InstallConfig, usingAgentMethod bool) field.
...
 	}
+
 	multiArchEnabled := types.MultiArchFeatureGateEnabled(c.Platform.Name(), c.EnabledFeatureGates())
 	allErrs = append(allErrs, validateCompute(&c.Platform, c.ControlPlane, c.Compute, field.NewPath("compute"), multiArchEnabled)...)
-	if multiArchEnabled {
-		allErrs = append(allErrs, validateMultiReleasePayload(c.ControlPlane, c.Compute)...)
+
+	releaseArch, err := version.ReleaseArchitecture()
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(field.NewPath(""), err))
```
Comment (rna-afk) [inline pkg/types/validation/installconfig.go:134]: Do we need an empty new path? wouldn't just field work

Comment (r4f4) [reply pkg/types/validation/installconfig.go]: Updated.

**Pull request merged**